### PR TITLE
enable custom path to proxy template file

### PR DESCRIPTION
**Pull request by kidiyoor**
wrt https://github.com/istio/istio/issues/3745

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -131,6 +131,9 @@ message ProxyConfig {
 
   // Address of the discovery service exposing xDS with plain text connection.
   string discovery_plain_address = 19;
+
+  // Path to the proxy template file
+  string proxy_template_path = 20;
```
Comment (ZackButcher) [inline mesh/v1alpha1/config.proto]: Lets indicate that this is the bootstrap config template - maybe `proxy_bootstrap_template_path`?

Comment (kidiyoor) [reply mesh/v1alpha1/config.proto]: yes. fixed. using proxy_bootstrap_template_path now

**Comment by ZackButcher** [comment]
No technical problem with this, just want to verify that `--custom-config-file` doesn't work for your use case first (see the comments in the issue).

**Comment by kidiyoor** [comment]
There should be a way for users to provide 'customized template' in a 'not hardcoded path'.

**Pull request merged**